src/client/storage.js

```diff
@@ -30,6 +32,88 @@ export function getSettings (key: string): StoredSettings {
     return JSON.parse('{}');
   }
 }
+
+function parseStartDate (date: ?string) {
+  if (!date) {
+    return undefined;
+  }
+  if (new Date(date).toString() === 'Invalid Date') {
+    return undefined;
+  }
+  return new Date(date);
+}
+function parseEndDate (date: ?string) {
+  if (!date) {
+    return undefined;
+  }
+  if (new Date(date).toString() === 'Invalid Date') {
+    return undefined;
+  }
+  if (date.split(' ').length === 1) {
+    return new Date(date + ' 23:59');
+  } else {
+    return new Date(date);
+  }
+}
+
+function encodeStartDate (date: ?Date) {
+  if (!date) {
+    return undefined;
+  }
+  const y = date.getFullYear();
+  const mon = date.getMonth() + 1;
+  const d = date.getDate();
+  const h = date.getHours();
+  const min = date.getMinutes();
+  if (h === 0 && min === 0) {
+    return `${y}-${mon}-${d}`;
+  } else {
+    return `${y}-${mon}-${d} ${h}:${min}`;
+  }
+}
+function encodeEndDate (date: ?Date) {
+  if (!date) {
+    return undefined;
+  }
+  const y = date.getFullYear();
+  const mon = date.getMonth() + 1;
+  const d = date.getDate();
```

Comment:
We're including `moment` in `package.json`.  Might as well [use it here](https://momentjs.com/docs/#/displaying/)